Build fails with "Found more than one jar in the 'lintChecks' configuration"

This happens after applying `id("com.soundcloud.delect")` (0.2.0) in root project's `build.gradle` file, and running the build with `./gradlew app:assembleDevDebug -Pdagger.reflect=true`. We have a mix of android-library and kotlin subprojects.

Output (`app` is the only application module in the project):
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:prepareLintJar'.
> Found more than one jar in the 'lintChecks' configuration. Only one file is supported. If using a separate Gradle project, make sure compilation dependencies are using compileOnly

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:prepareLintJar'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:187)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:185)
```

--- a/buildSrc/src/main/java/com/soundcloud/reflect/DaggerReflectPlugin.kt
+++ b/buildSrc/src/main/java/com/soundcloud/reflect/DaggerReflectPlugin.kt
@@ -32,7 +32,8 @@ class DaggerReflectPlugin : Plugin<Project> {
                                 this@config.name,
                                 "$reflectDaggerGroupId:dagger-reflect:${extension.daggerReflectVersion}"
                             )
-                            if (maxUnsupportedDaggerLintVersion < VersionNumber.parse(extension.daggerReflectVersion)) {
+                            if (extension.enableReflectLintCheck &&
+                                (maxUnsupportedDaggerLintVersion < VersionNumber.parse(extension.daggerReflectVersion))) {
                                 whenLintPluginAdded {
                                     add(
                                         "lintChecks",

--- a/buildSrc/src/main/java/com/soundcloud/reflect/DelectExtension.kt
+++ b/buildSrc/src/main/java/com/soundcloud/reflect/DelectExtension.kt
@@ -11,4 +11,6 @@ open class DelectExtension {
      * https://github.com/jakewharton/dagger-reflect#full-reflection
      */
     var addReflectAnnotationProcessor = true
+
+    var enableReflectLintCheck = true
 }

